torch/_inductor/ir.py

```diff
@@ -5261,3 +5262,97 @@ def codegen_collective(self, wrapper, output_name, input_names):
             f"group={output_name}_pg, "
             "async_op=True)"
         )
+
+
+class AllToAllSingle(OutOfPlaceCollectiveKernel):
+    count = itertools.count()
+
+    def __init__(
+        self, layout, inputs, outputs, constant_args,
+        inputs_allow_none,
+    ):
+        super().__init__(layout, inputs, outputs, constant_args)
+        self.inputs_allow_none = inputs_allow_none
+
+    @classmethod
+    def create(
+        cls,
+        x: "TensorBox",
+        output_split_sizes: Optional["TensorBox"],
+        input_split_sizes: Optional["TensorBox"],
+        tag: str,
+        ranks: List[int],
+        group_size: int,
+    ):
+        x_realized = cls.realize_input(x)
+        inputs_allow_none = []
+        output_split_sizes_realized = None
+        if output_split_sizes is not None:
+            output_split_sizes_realized = cls.realize_input(output_split_sizes)
+            # NCCL/GLOO alltoall_base also assumes output_split_sizes to be int64,
+            # so hardcoding int64 here should be okay.
+            V.graph.name_to_coerced_dtype[output_split_sizes_realized.data.name] = torch.int64
```

Comment:
Yeah looking at the setter site, I also still don't understand what this means

Reply:
we no longer need this